app/controllers/nonsipp/otherassetsdisposal/ReportedOtherAssetsDisposalListController.scala

```diff
@@ -203,29 +286,67 @@ object ReportedOtherAssetsDisposalListController {
 
   def viewModel(
     srn: Srn,
+    mode: Mode,
     page: Int,
     disposals: Map[Max5000, List[Max50]],
-    userAnswers: UserAnswers
+    userAnswers: UserAnswers,
+    viewOnlyUpdated: Boolean,
+    optYear: Option[String] = None,
+    optCurrentVersion: Option[Int] = None,
+    optPreviousVersion: Option[Int] = None,
+    compilationOrSubmissionDate: Option[LocalDateTime] = None
   ): FormPageViewModel[ListViewModel] = {
 
     val numberOfDisposals = disposals.map { case (_, disposalIndexes) => disposalIndexes.size }.sum
     val numberOfOtherAssetsItems = userAnswers.map(OtherAssetsCompleted.all(srn)).size
     val maxPossibleNumberOfDisposals = maxDisposalPerOtherAsset * numberOfOtherAssetsItems
 
-    val (title, heading) = if (numberOfDisposals == 1) {
-      ("assetDisposal.reportedOtherAssetsDisposalList.title", "assetDisposal.reportedOtherAssetsDisposalList.heading")
-    } else {
-      (
-        "assetDisposal.reportedOtherAssetsDisposalList.title.plural",
-        "assetDisposal.reportedOtherAssetsDisposalList.heading.plural"
-      )
+//    val (title, heading) = if (numberOfDisposals == 1) {
+//      ("assetDisposal.reportedOtherAssetsDisposalList.title", "assetDisposal.reportedOtherAssetsDisposalList.heading")
+//    } else {
+//      (
+//        "assetDisposal.reportedOtherAssetsDisposalList.title.plural",
+//        "assetDisposal.reportedOtherAssetsDisposalList.heading.plural"
+//      )
+//    }
+
+    val (title, heading) = ((mode, numberOfDisposals) match {
+      case (ViewOnlyMode, _) if numberOfDisposals > 1 =>
+        (
+          "assetDisposal.reportedOtherAssetsDisposalList.view.title.plural",
+          "assetDisposal.reportedOtherAssetsDisposalList.view.heading.plural"
+        )
+      case (ViewOnlyMode, numberOfDisposals) if numberOfDisposals > 1 =>
+        (
+          "assetDisposal.reportedOtherAssetsDisposalList.view.title",
+          "assetDisposal.reportedOtherAssetsDisposalList.view.heading"
+        )
+      case (ViewOnlyMode, numberOfDisposals) if numberOfDisposals > 1 =>
```

Comment:
It looks like this case is identical to the one above but with different content - is this case supposed to be for non-`ViewOnlyMode`?